### ISSUE TITLE
memorySize of the lambdas should be 128.

### ISSUE THREAD

**Issue by teebu** [alexcasalboni/aws-lambda-power-tuning#11]
There is no reason for the tester functions to be so big,

```
functions:
  initializer:
    handler: lambda/initializer.handler
    memorySize: 128
    timeout: 60
  executor:
    handler: lambda/executor.handler
    memorySize: 128
    timeout: 300
  cleaner:
    handler: lambda/cleaner.handler
    memorySize: 128
    timeout: 60
  finalizer:
    handler: lambda/finalizer.handler
    memorySize: 128
    timeout: 60
```

**Comment by alexcasalboni** [comment]
Hi @teebu, that's right. I didn't spend time optimizing the state machine functions too. My thinking was that they don't really need to scale up, and therefore the cost of each execution is not worth optimizing.

I'm pretty sure that your suggested config is great for `initializer`, `cleaner` and `finalizer`, while it might still make sense to have higher memory/performance for the `executor`.

**Comment by teebu** [comment]
`executor` just invokes lambdas, doesn't do much else, 128 seemed enough for what it does.

**Comment by alexcasalboni** [comment]
Let's not forget the more RAM also means more compute power and faster network.

In this case, faster network might speed things up a lot since every AWS Lambda invocation is in fact a call over the network (even if not much data is sent around). Of course, this should be measured and properly optimized.

If you run everything in series and each invocation takes tens of seconds, you won't notice much difference. But if you run everything in parallel and each invocation takes less than a second, the effect/delay of network calls will be relevant.

**Comment by teebu** [comment]
I did a test on a small function, a simple hello world with a delay of 3000 ms, with executioner set to 128:

```
{
  "lambdaARN": "arn:aws:lambda:us-east-1:xxxx:function:hello-world",
  "num": 5,
  "parallelInvocation": true,
  "payload": {
    "key1": "abc"
  }
}
```

```
[
  {
    "cheapest": {
      "power": "1024",
      "cost": 0.000051584,
      "duration": 3007.4599999999996
    },
    "fastest": {
      "power": "1556",
      "cost": 0.0000783835,
      "duration": 3006.586666666666
    },
    "stats": [
      {
        "power": "1024",
        "cost": 0.000051584,
        "duration": 3007.4599999999996
      },
      {
        "power": "1556",
        "cost": 0.0000783835,
        "duration": 3006.586666666666
      },
      {
        "power": "2048",
        "cost": 0.000103168,
        "duration": 3007.216666666667
      }
    ]
  },
  {}
]
```


However, a bigger function that takes 200,000ms+ fails with executioner taking too long, times out after 300,000ms. Changing it to 1028 doesn't seem to fix it. Haven't spent much time researching why, but I don't think size of executioner made a difference.

**Comment by alexcasalboni** [comment]
Agreed, that wouldn't solve your long-execution problem.

Did you also record how long the state machine took to complete?

Using more "power" might reduce the total state machine execution time for short-running functions, but I don't have metrics to show at the moment.

Anyways, after you've collected more data and proved that using 128MB of RAM doesn't slow down the overall execution, feel free to open a PR and I'll be happy to review your changes and your stats 😄 🚀

**Comment by alexcasalboni** [comment]
Fixed in #16 🎉
